nengo/params.py

```diff
@@ -130,8 +132,29 @@ def get_default(self, obj):
     def set_default(self, obj, value):
         if not self.configurable:
             raise ConfigError("Parameter '%s' is not configurable" % self)
-        self.validate(obj, value)
-        self._defaults[obj] = value
+        self._defaults[obj] = self.coerce(obj, value)
+
+    def check_type(self, instance, value, type_):
+        if value is not None and not isinstance(value, type_):
+            if isinstance(type_, tuple):
+                type_str = " or ".join((t.__name__ for t in type_))
+            else:
+                type_str = type_.__name__
+            raise ValidationError("Must be of type %r (got type %r)."
+                                  % (type_str, type(value).__name__),
+                                  attr=self.name, obj=instance)
+
+    def coerce(self, instance, value):
+        if isinstance(value, DefaultType):
+            raise ValidationError("Default is not a valid value. To reset a "
+                                  "parameter, use 'del'.",
+                                  attr=self.name, obj=instance)
+        if self.readonly and instance in self.data:
```

Comment:
I may have brought this up before, but when there's a bunch of chained errors, I feel like they should be `elif` instead of just `if`. I think the conclusion we came to last time was that an `elif` implies priority when there is none in this case?